src/test/java/com/synacy/lesson01/exercises/ProductTest.java

```diff
@@ -0,0 +1,14 @@
+package com.synacy.lesson01.exercises;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class ProductTest {
+
+    @Test(expected = IllegalArgumentException.class)
+    public void setUnitPrice_shouldThrowIllegalArgumentExceptionWhenUnitPriceIsLessThanZero() throws Exception {
+        Product product = new Product("new product", -1.0);
```

Comment:
Test method name does not match assertion.

The test method gives me the idea that you are testing the `setUnitPrice` method but no method invocation but the constructor was made. You could just have made changed the name  to
```
"constructor_shouldThrowIllegalArgumentExceptionWhenUnitPriceIsLessThanZero()"
```